src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -28,83 +34,42 @@ protected override bool SupportsPropertyInitializer(Compilation compilation)
             }
+
+            foreach (var memberAccess in codeBlock.DescendantNodesAndSelf().OfType<MemberAccessExpressionSyntax>())
+                AddIneligibleFieldsIfAccessedOffNotDefinitelyAssignedValue(semanticModel, memberAccess, ineligibleFields, cancellationToken);
         }
 
-        protected override void RegisterIneligibleFieldsAction(
-            List<AnalysisResult> analysisResults, HashSet<IFieldSymbol> ineligibleFields,
-            Compilation compilation, CancellationToken cancellationToken)
+        protected override void RegisterNonConstructorFieldWrites(
+            ConcurrentDictionary<IFieldSymbol, ConcurrentSet<SyntaxNode>> fieldWrites, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken)
         {
-            var groups = analysisResults.Select(r => (typeDeclaration: (TypeDeclarationSyntax)r.PropertyDeclaration.Parent!, r.SemanticModel))
-                                        .Distinct()
-                                        .GroupBy(n => n.typeDeclaration.SyntaxTree);
-
-            foreach (var (tree, typeDeclarations) in groups)
-            {
-                foreach (var (typeDeclaration, semanticModel) in typeDeclarations)
-                {
-                    foreach (var argument in typeDeclaration.DescendantNodesAndSelf().OfType<ArgumentSyntax>())
-                    {
-                        // An argument will disqualify a field if that field is used in a ref/out position.  
-                        // We can't change such field references to be property references in C#.
-                        if (argument.RefKindKeyword.Kind() != SyntaxKind.None)
-                            AddIneligibleFields(semanticModel, argument.Expression, ineligibleFields, cancellationToken);
-                    }
-
-                    foreach (var refExpression in typeDeclaration.DescendantNodesAndSelf().OfType<RefExpressionSyntax>())
-                        AddIneligibleFields(semanticModel, refExpression.Expression, ineligibleFields, cancellationToken);
-
-                    // Can't take the address of an auto-prop.  So disallow for fields that we do `&x` on.
-                    foreach (var addressOfExpression in typeDeclaration.DescendantNodesAndSelf().OfType<PrefixUnaryExpressionSyntax>())
-                    {
-                        if (addressOfExpression.Kind() == SyntaxKind.AddressOfExpression)
-                            AddIneligibleFields(semanticModel, addressOfExpression.Operand, ineligibleFields, cancellationToken);
-                    }
-
-                    foreach (var memberAccess in typeDeclaration.DescendantNodesAndSelf().OfType<MemberAccessExpressionSyntax>())
-                        AddIneligibleFieldsIfAccessedOffNotDefinitelyAssignedValue(semanticModel, memberAccess, ineligibleFields, cancellationToken);
-                }
-            }
+            // nothing to do here for C#.  This is for VB only situations.
         }
```

Comment:
VB had some funky logic that was forcing us to go back to processing other files (and getting their semantic models).  refactored to make no longer necessary.